Seldon add ssl (#3813)
SeldonIO/seldon-core: doc/source/analytics/logging.md

 * LOGGER_KAFKA_BROKER : The Kafka Broker service endpoint.
 * LOGGER_KAFKA_TOPIC : The kafka Topic to log the requests.

### Logging to encrypted Kafka with SSL

You can log requests to an encrypted Kafka with SSL. SSL uses private-key/ certificate pairs, which are used during the SSL handshake process. 

> To be able to log payload, the client needs to authenticate with SSL, the client needs its own keystore, made up of a key pair and a signed certificate, signed by the CA certificate file for verifying the broker's certificate. Here is an example below on how to define those for a deployment: 

Reviewer: :thought_balloon: It might be nice to make this a bit clearer with a list, e.g.
```suggestion
To be able to log payloads, the client needs:
* to authenticate with SSL
* its own keystore, made up of a key pair and a signed certificate
* the CA certificate used to sign the key-certificate pair

The CA certificate needs to be recognised by the broker and can also be used for verifying the broker's certificate.

Here is an example on how to define these for a deployment: 
```

Reviewer: :thought_balloon: It would also be quite nice to link to the Confluent Kafka documentation in case users need any further details

Author: Do you have [that documentation](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html#clients) in mind or another one?

Reviewer: The [overall page](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html) is helpful, but it doesn't really talk about the options we're exposing here, so probably also good to link through to the [librdkafka](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) config properties, as that's how to properly understand what's on offer and what values to provide.

I'll admit it's a bit detailed and implementation-specific, but it's useful.

Author: Done! :D

```yaml
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  name: cifar10
  namespace: seldon
spec:
  name: resnet32
  predictors:
  - graph:
      implementation: TRITON_SERVER
      logger:
        mode: all
      modelUri: gs://seldon-models/triton/tf_cifar10
      name: cifar10
    name: default
    svcOrchSpec:
      env:
      - name: LOGGER_KAFKA_BROKER
        value: seldon-kafka-plain-0.kafka:9092
      - name: LOGGER_KAFKA_TOPIC
        value: seldon
      - name: KAFKA_SECURITY_PROTOCOL
        value: ssl
      - name: KAFKA_SSL_CA_CERT_FILE
        value: /path/to/ca.pem
      - name: KAFKA_SSL_CLIENT_CERT_FILE

>         value: /path/to/ca.cert

Reviewer: :upside_down_face: Minor suggestion, but would recommend not calling the file `ca` to avoid confusion with the actual CA cert - perhaps `access.cert`?

      - name: KAFKA_SSL_CLIENT_KEY_FILE
        value: /path/to/access.key
      - name: KAFKA_SSL_CLIENT_KEY_PASS

>         value: abcdefgh # Key password, if any (optional field)

Reviewer: :thought_balloon: As these docs are de facto the recommended way of doing things, it'd be nicer to use the field reference syntax here:
```yaml
valueFrom:
  secretKeyRef:
    name: my-kafka-secret
    key: ssl-password
```
It's normally not a good idea to put secrets in plain text into resources or anywhere else!

The comment is good to include, though :slightly_smiling_face:

Author: Ah yeah good point, I wouldn't do it but I kinda forgot about security when I wrote the documentation 😬

 * LOGGER_KAFKA_BROKER : The Kafka Broker service endpoint.
 * LOGGER_KAFKA_TOPIC : The kafka Topic to log the requests.

### Logging to encrypted Kafka with SSL

You can log requests to an encrypted Kafka with SSL. SSL uses private-key/ certificate pairs, which are used during the SSL handshake process. 

To be able to log payloads, the client needs:
* to authenticate with SSL
* its own keystore, made up of a key pair and a signed certificate
* the CA certificate used to sign the key-certificate pair

> The CA certificate needs to be recognised by the broker and can also be used for verifying the broker's certificate. It is possible to read more about the different options available on [Confluent documentation](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html) and [librdkafka Configuration](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) page. 

Reviewer: ```suggestion
The CA certificate needs to be recognised by the broker and can also be used for verifying the broker's certificate.

It is possible to read more about the different options available on the [Confluent documentation](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html) and [librdkafka Configuration](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) pages. 
```
Just to separate the configuration details from the general setup steps

Here is an example on how to define these for a deployment: 

```yaml
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  name: cifar10
  namespace: seldon
spec:
  name: resnet32
  predictors:
  - graph:
      implementation: TRITON_SERVER
      logger:
        mode: all
      modelUri: gs://seldon-models/triton/tf_cifar10
      name: cifar10
    name: default
    svcOrchSpec:
      env:
      - name: LOGGER_KAFKA_BROKER
        value: seldon-kafka-plain-0.kafka:9092
      - name: LOGGER_KAFKA_TOPIC
        value: seldon
      - name: KAFKA_SECURITY_PROTOCOL
        value: ssl
      - name: KAFKA_SSL_CA_CERT_FILE
        value: /path/to/ca.pem
      - name: KAFKA_SSL_CLIENT_CERT_FILE
        value: /path/to/access.cert
      - name: KAFKA_SSL_CLIENT_KEY_FILE
        value: /path/to/access.key
      - name: KAFKA_SSL_CLIENT_KEY_PASS
        valueFrom:
          secretKeyRef:
            name: my-kafka-secret
            key: ssl-password # Key password, if any (optional field)
    replicas: 1
  protocol: kfserving

```

> =

Reviewer: :upside_down_face: Minor typo
```suggestion

```
